benchmarks/project/src/main/java/io/zeebe/Starter.java

```diff
@@ -59,30 +62,39 @@ public void run() {
     LOG.info("Creating an instance every {}ms", intervalMs);
 
     final String variables = readVariables(starterCfg.getPayloadPath());
+    final LocalDateTime startTime = LocalDateTime.now();
     executorService.scheduleAtFixedRate(
         () -> {
-          try {
-            if (starterCfg.isWithResults()) {
-              requestFutures.put(
-                  client
-                      .newCreateInstanceCommand()
-                      .bpmnProcessId(processId)
-                      .latestVersion()
-                      .variables(variables)
-                      .withResult()
-                      .requestTimeout(starterCfg.getWithResultsTimeout())
-                      .send());
-            } else {
-              requestFutures.put(
-                  client
-                      .newCreateInstanceCommand()
-                      .bpmnProcessId(processId)
-                      .latestVersion()
-                      .variables(variables)
-                      .send());
+          final long duration = ChronoUnit.SECONDS.between(startTime, LocalDateTime.now());
+          if (durationLimit <= 0 || duration < durationLimit) {
+            try {
+              if (starterCfg.isWithResults()) {
+                requestFutures.put(
+                    client
+                        .newCreateInstanceCommand()
+                        .bpmnProcessId(processId)
+                        .latestVersion()
+                        .variables(variables)
+                        .withResult()
+                        .requestTimeout(starterCfg.getWithResultsTimeout())
+                        .send());
+              } else {
+                requestFutures.put(
+                    client
+                        .newCreateInstanceCommand()
+                        .bpmnProcessId(processId)
+                        .latestVersion()
+                        .variables(variables)
+                        .send());
+              }
+            } catch (Exception e) {
+              LOG.error("Error on creating new workflow instance", e);
             }
-          } catch (Exception e) {
-            LOG.error("Error on creating new workflow instance", e);
+          } else {
+            // TODO can one use scheduledFuture.cancel(false) to gracefully
```

Comment:
Who is the one ? :sweat_smile:

Reply:
So what you want to do is canceling the timer, stopping the execution service. Await all remaining responses and then exit right?